hw/ip/spi_host/doc/theory_of_operation.md

```diff
@@ -1,87 +1,71 @@
 # Theory of Operation
 
-# SPI Protocol Basics
+- [SPI Protocol Basics](#spi-protocol-basics)
+- [API Details](#api-details): *Interface information, such as models for conceptual interactions, CSRs and interrupts.*
```

Comment:
Nit: Are the italics really necessary here? I'd consider it without the stars. (But I definitely don't feel strongly about this!)

Reply:
See the following screenshot, I thought it useful to be visually distinct from the links themselves
![image](https://github.com/lowRISC/opentitan/assets/102029880/95fa81d8-cffb-4c3e-a1b2-574181fc2277)